broker/data-access-layer/iaas/src/AwsClient.js

```diff
@@ -0,0 +1,308 @@
+'use strict';
+
+const _ = require('lodash');
+const Promise = require('bluebird');
+const S3 = require('aws-sdk/clients/s3');
+const logger = require('@sf/logger');
+const {
+  CONST,
+  errors: {
+    NotFound,
+    Unauthorized
+  },
+  commonFunctions: {
+    streamToPromise
+  }
+} = require('@sf/common-utils');
+
+const ComputeClient = require('./ComputeClient');
+const BaseCloudClient = require('./BaseCloudClient');
+
+class AwsClient extends BaseCloudClient {
+  constructor(settings) {
+    super(settings);
+    this.constructor.validateParams(this.settings);
+    this.storage = this.constructor.createStorageClient(_
+      .chain(this.settings)
+      .omit('name')
+      .set('provider', this.provider)
+      .value()
+    );
+    this.blockstorage = this.constructor.createComputeClient(_
+      .chain(this.settings)
+      .set('provider', this.provider)
+      .value()
+    );
+    this.storage.listObjectsAsync = Promise.promisify(this.storage.listObjects, {});
+    this.storage.deleteObjectAsync = Promise.promisify(this.storage.deleteObject, {});
+    this.storage.uploadAsync = Promise.promisify(this.storage.upload, {});
+  }
+
+  createDiskFromSnapshot(snapshotId, zones, options = {}) {
+    let tags = [];
+    if (options.tags) {
+      options.tags.createdBy = 'service-fabrik';
+    } else {
+      options.tags = {
+        createdBy: 'service-fabrik'
+      };
+    }
+    tags.push({
+      ResourceType: 'volume',
+      Tags: _.keys(options.tags).map(tagKey => ({
+        Key: tagKey,
+        Value: options.tags[tagKey]
+      }))
+    });
+    return Promise.try(() => {
+      return this.blockstorage
+        .createVolume({
+          AvailabilityZone: _.isArray(zones) ? zones[0] : zones,
+          SnapshotId: snapshotId,
+          VolumeType: _.get(options, 'type', 'gp2'),
+          TagSpecifications: tags
+        })
+        .promise();
+    })
+      .then(volume => {
+        const describeReq = {
+          VolumeIds: [volume.VolumeId]
+        };
+        return this.blockstorage.waitFor('volumeAvailable', describeReq).promise();
+      })
+      .then(volResponse => volResponse.Volumes[0])
+      .then(volume => {
+        const responseTags = volume.Tags || [];
+        const outTags = {};
+        _.forEach(responseTags, tag => {
+          outTags[tag.Key] = tag.Value;
+        });
+        return {
+          volumeId: volume.VolumeId,
+          size: volume.Size,
+          zone: volume.AvailabilityZone,
+          type: volume.VolumeType,
+          extra: {
+            type: volume.VolumeType,
+            tags: outTags
+          }
+        };
+      });
+  }
+
+  getDiskMetadata(diskId) {
```

Comment:
Involves operation with respect to `computeClient`, so directly took from `CloudProviderClient.js`